odoo/models.py

```diff
@@ -3529,37 +3530,52 @@ def write(self, vals):
               :const:`odoo.tools.misc.DEFAULT_SERVER_DATETIME_FORMAT`
         * .. _openerp/models/relationals/format:
 
-          :class:`~odoo.fields.One2many` and
-          :class:`~odoo.fields.Many2many` use a special "commands" format to
-          manipulate the set of records stored in/associated with the field.
-
-          This format is a list of triplets executed sequentially, where each
-          triplet is a command to execute on the set of records. Not all
-          commands apply in all situations. Possible commands are:
-
-          ``(0, 0, values)``
-              adds a new record created from the provided ``value`` dict.
-          ``(1, id, values)``
-              updates an existing record of id ``id`` with the values in
-              ``values``. Can not be used in :meth:`~.create`.
-          ``(2, id, 0)``
-              removes the record of id ``id`` from the set, then deletes it
-              (from the database). Can not be used in :meth:`~.create`.
-          ``(3, id, 0)``
-              removes the record of id ``id`` from the set, but does not
-              delete it. Can not be used in
-              :meth:`~.create`.
-          ``(4, id, 0)``
-              adds an existing record of id ``id`` to the set.
-          ``(5, 0, 0)``
-              removes all records from the set, equivalent to using the
-              command ``3`` on every record explicitly. Can not be used in
-              :meth:`~.create`.
-          ``(6, 0, ids)``
-              replaces all existing records in the set by the ``ids`` list,
-              equivalent to using the command ``5`` followed by a command
-              ``4`` for each ``id`` in ``ids``.
-        """
+          The expected value of a :class:`~odoo.fields.One2many` or
+          :class:`~odoo.fields.Many2many` relational field is a list of
+          :class:`~odoo.fields.Command` that manipulate the relation they
+          implement. There are a total of 7 commands:  create, update, delete,
+          unlink, link, and set. Not all commands apply in all situations.
+
+          * The ``create(vals)`` command (id: 0) is used to create new records
+            in the comodel using ``vals`` and to link them to ``self``.
+
+            In case of a **Many2many** relation, one unique new record is
+            created in the comodel such that all records in `self` are linked to
+            the new record.
+
+            In case of a **One2many** relation, one new record is created in the
+            comodel for every record in ``self`` such that every record in
+            ``self`` is linked to exactly one of the new records.
+
+          * The ``update(id, vals)`` command (id: 1) is used to write ``vals``
```

Comment:
Using `id` for the command identifier is very confusing here (and in Odoo in general). "code", "variant", "discriminant", "tag", ... would be better alternatives.

Also maybe these could ref' the relevant `Command` method? That way we'd get a nice hyperlink and the formatting ought be correct?

Reply:
Here I was using ref in the general sense, you'd want to use the `:meth:` role from the Python domain: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#role-py-meth (under the cover autodoc should simply generate the relevant [python domain structure](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-python-domain))

Reply:
Thank you, that's what I ended up using. The long documentation is now at Command under each method. The write documentation just enumerates the various command and link them using :meth:, I find it better this way